### PR TITLE
Fix duplicate env in loader for pserve, ...

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: c2cwsgiutils/loader.py [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
+LOG = logging.getLogger(__name__)
+
 
 class Loader(BaseLoader):  # type: ignore
     """The application loader."""
 
     def _get_defaults(self, defaults: Optional[Dict[str, str]] = None) -> Dict[str, str]:
         d: Dict[str, str] = {}
-        d.update(os.environ)
+        lowercase_keys: Set[str] = set()
```
Comment (arnaud-morvan) [inline c2cwsgiutils/loader.py:15]: May desserve a comment:
```
# configparser does not support duplicated defaults with different cases.
```

File: tests/test_loader.py [hunk 2]
```diff
@@ -18,3 +18,19 @@ def test_loader():
...
+[app:main]
+variable = %(VARIABLE)s
+"""
+    ),
+)
+@patch.dict("c2cwsgiutils.loader.os.environ", {"VARIABLE": "value", "variable": "value"})
+def test_loader_dbl():
+    loader = get_config_loader("c2c:///app/production.ini")
+    assert 'c2cwsgiutils.loader.Loader(uri="c2c:///app/production.ini")' == repr(loader)
+    assert "value" == loader._get_defaults()["VARIABLE"]  # pylint: disable=W0212
```
Comment (arnaud-morvan) [inline tests/test_loader.py:36]: test name is not enough expressive, I can propose: `test_loader_with_duplicates_in_env`
and to be able to run each test individually I would rename the first one: `test_loader_success`

Your test pass without the changes in loader. Maybe you can add:
```
assert "variable" not in loader._get_defaults()
```

**Review by arnaud-morvan — APPROVED** [review]

**Comment by arnaud-morvan** [comment]
@sbrunner : It seems this project contains:
/acceptance_tests/tests
/tests

And we can suppose in makefile that we run tests on /tests folder, but it runs the ones from /acceptance_tests/tests
And it seems that the ones from /tests are not runned.

**Comment by arnaud-morvan** [comment]
found: tests in /tests are runned by `make build_docker_tests`

**Comment by sbrunner** [comment]
@arnaud-morvan I agree that not very convignent...

**Pull request merged**